resources/server/db/db_wrapper.ts

```diff
@@ -1,21 +1,28 @@
 import { pool } from './pool';
 import { mainLogger } from '../sv_logger';
-import { ResultSetHeader } from 'mysql2';
+import { ResultSetHeader, RowDataPacket } from 'mysql2';
 import { config } from '../server';
-import { withProfile } from '../utils/withProfile';
-
-const RESOURCE_NAME = GetCurrentResourceName();
 
 // Replicates ghmattimysql API
 class _DbInterface {
   private readonly logger = mainLogger.child({ module: 'DBInterface' });
 
   private async _internalQuery(query: string, values?: unknown) {
     try {
-      if (config.database.profileQueries) return await withProfile(pool.execute, query, values);
+      const conn = await pool.getConnection();
+
+      if (config.database.profileQueries) {
+        const startTime = process.hrtime();
+        const result = await conn.execute(query, values);
+        this.logger.info(
+          `${query} (${values}) execution time: ${process.hrtime(startTime)[1] / 1000000}ms`,
+        );
+        return result;
+      }
 
-      ScheduleResourceTick(RESOURCE_NAME);
-      return await pool.query(query, values);
+      const result = await conn.execute(query, values);
+      conn.release();
```

Comment:
Whats the reason for managing the connection release manually? Isn't this wrapped internally within `mysql` query wrapper?